src/binary/ibc-setup/commands/init.spec.ts

```diff
@@ -0,0 +1,89 @@
+import fs from 'fs';
+
+import test from 'ava';
+import axios from 'axios';
+import sinon from 'sinon';
+
+import { Options, run } from './init';
+
+const consoleLog = sinon.stub(console, 'log');
+const fsExistSync = sinon.stub(fs, 'existsSync');
+const fsMkdirSync = sinon.stub(fs, 'mkdirSync');
+const axiosGet = sinon.stub(axios, 'get');
+const fsReadFileSync = sinon.stub(fs, 'readFileSync');
+const fsWriteFileSync = sinon.stub(fs, 'writeFileSync');
+const fsLstatSync = sinon.stub(fs, 'lstatSync');
+
+test.beforeEach(() => {
+  sinon.reset();
+
+  fsLstatSync.returns({
+    ...new fs.Stats(),
+    isDirectory: () => true,
+    isFile: () => true,
+  });
+});
+
+test('read existing registry.yaml', async (t) => {
+  const options: Options = {
+    home: '/home/user',
+    src: 'AAA',
+    dest: 'BBB',
+  };
+
+  fsExistSync.onCall(0).returns(true).onCall(1).returns(true);
+
+  const registryPath = `${options.home}/registry.yaml`;
+  const registryYaml = `
+  version: 1
+  `;
+
+  axiosGet.resolves({
+    data: registryYaml,
+  });
+  fsReadFileSync.returns(registryYaml);
+  fsWriteFileSync.returns(undefined);
```

Comment:
Can we group all the test fixtures together and all the stubbing together?

Reply:
55bbb32